Q-9) Test the hypothesis that the mean scores of all the students are the same. If not, name the student with the highest score

In [1]:
import numpy as np
from scipy.stats import f_oneway


#F test
exam_1 = [85, 70, 90, 75, 95]
exam_2 = [90, 80, 85, 70, 92]
full_exam = [92, 85, 88, 75, 96]

f_value, p_value = f_oneway(exam_1, exam_2, full_exam)





print("F-value:", f_value)
print("F critical value from table is 8.51")



F-value: 0.3236947791164659
F critical value from table is 8.51


Since the F-value (0.323694779) is less than 8.51, we fail to reject the null hypothesis.Therefore, there is not enough evidence to conclude that the mean scores of all the students are significantly different.

and Jeevan has the highest scores in all three exams with scores of 95, 92, and 96 in Exam 1, Exam 2, and Full Exam